### PR TITLE
CDRIVER-2744: Allow primary RP in _mongoc_cursor_new_with_opts()

### PR DESCRIPTION
**Pull request by jmikola**
https://jira.mongodb.org/browse/CDRIVER-2744

### CODE REVIEW

**Comment by ajdavis** [comment]
Would you mind adding a basic C Driver test in test-mongoc-transactions.c?

File: src/libmongoc/tests/test-mongoc-transactions.c [hunk 1]
```diff
@@ -356,6 +356,47 @@ test_unknown_commit_result (void)
...
+   read_prefs = mongoc_read_prefs_new (MONGOC_READ_PRIMARY);
+
+   cursor = mongoc_collection_find_with_opts (
+      collection, tmp_bson ("{}"), &opts, read_prefs);
+
+   bson_destroy (&opts);
+   mongoc_read_prefs_destroy (read_prefs);
+   mongoc_collection_destroy (collection);
+
+   ASSERT_OR_PRINT (!mongoc_cursor_error (cursor, &error), error);
```
Comment (ajdavis) [inline src/libmongoc/tests/test-mongoc-transactions.c]: Let's iterate the cursor to completion first; this catches the current bug but I could easily imagine a new one that doesn't manifest until mongoc_cursor_next.

File: src/libmongoc/tests/test-mongoc-transactions.c [hunk 2]
```diff
@@ -356,6 +356,49 @@ test_unknown_commit_result (void)
...
+   read_prefs = mongoc_read_prefs_new (MONGOC_READ_PRIMARY);
+
+   cursor = mongoc_collection_find_with_opts (
+      collection, tmp_bson ("{}"), &opts, read_prefs);
+
+   bson_destroy (&opts);
+   mongoc_read_prefs_destroy (read_prefs);
+   mongoc_collection_destroy (collection);
+
+   ASSERT (!mongoc_cursor_next (cursor, &doc));
```
Comment (jmikola) [inline src/libmongoc/tests/test-mongoc-transactions.c]: > Let's iterate the cursor to completion first

Done.

**Comment by ajdavis** [comment]
Thanks! I'll cherry-pick to r1.11

**Pull request merged**